### ISSUE TITLE
xychart X-axis labels overlapping each other to be unreadable - need feature to rotate them by 45 degrees (GNUplot can do this)

### ISSUE THREAD

**Issue by HariSekhon** [mermaid-js/mermaid#5926]
### Description

xychart X-axis labels overlap with each other

Need a way to rotate them by 45 degrees to fit them better like GNUplot can do.

### Steps to reproduce

Code is here:

https://raw.githubusercontent.com/HariSekhon/Diagrams-as-Code/refs/heads/master/git_commits_per_month.mmd

Compare the same graph and data generated in GNUplot vs MermaidJS xychart with both charts one above the other here:

https://github.com/HariSekhon/Diagrams-as-Code/?tab=readme-ov-file#git-commits-per-month-for-this-repo

### Screenshots

![](https://raw.githubusercontent.com/HariSekhon/Diagrams-as-Code/master/images/git_commits_per_month.svg)

### Code Sample

```text
Code is here:

https://raw.githubusercontent.com/HariSekhon/Diagrams-as-Code/refs/heads/master/git_commits_per_month.mmd
```

### Setup

- Mermaid CLI version: 11.2.0
- Browser and Version: Chrome

### Suggested Solutions

Add ability to rotate X-axis labels like GNUplot can do which is seen in the chart at the link given above.

**Comment by criccomini** [comment]
Yes, please! We hit this, and ended up having to rotate our charts 90 degrees, which makes the data harder to understand.

**Comment by uiolee** [comment]
same

**Comment by ashishjain0512** [comment]
Approved
